romanesco/plugins/scala/executor.py

```diff
@@ -1,38 +1,40 @@
-import re
 import romanesco.utils
-import tempfile
+import os
+import json
 
 
-def _expand_args(args, inputs, taskInputs, tmpDir):
-    """
-    Expands arguments to the container execution if they reference input
-    data. For example, if an input has id=foo, then a container arg of the form
-    $input{foo} would be expanded to the runtime value of that input.
-    """
-    newArgs = []
-    regex = re.compile(r'\$input\{([^}]+)\}')
+def _write_scala_script(script, inputs, task_outputs, tmp_dir):
+    script_fname = os.path.join(tmp_dir, 'script.scala')
+    with open(script_fname, 'w') as script_file:
+        # Send input values to the script
+        for name, binding in inputs.iteritems():
+            value = json.dumps(binding['script_data'])
+            script_file.write('val ' + name + ' = ' + value + '\n')
 
-    for arg in args:
-        for inputId in re.findall(regex, arg):
-            if inputId in inputs:
-                arg = arg.replace('$input{%s}' % inputId,
-                                  inputs[inputId]['script_data'])
-            elif inputId == '_tempdir':
-                arg = arg.replace('$input{_tempdir}', tmpDir)
+        # Run the script
+        script_file.write(script)
 
-        newArgs.append(arg)
+        # Write output values to temporary files
+        script_file.write('import java.io._\n')
+        for name in task_outputs:
+            if name != '_stderr' and name != '_stdout':
+                fname = os.path.join(tmp_dir, name)
+                script_file.write("""
+new PrintWriter({}) {{
```

Comment:
> This is simply a placeholder for the python format function - it will be replaced by the name of the temporary file to be written to.

Ah.. makes sense.. thanks.